### DISCUSSION TITLE
Only PE After System install Visual Studio 2022

### DISCUSSION THREAD

**Discussion by Sicarmon** [dnSpyEx/dnSpy#362] [Q&A]
![image](https://github.com/user-attachments/assets/83f75f73-70c7-494b-8389-edadf6e059f2)

I was able to successfully decompile a .NET 4.0 exe file using dnspy v6.5.1, and it worked great. However, after I installed Visual Studio 2022, it stopped working properly. When I open the exe file, I only see the PE.

**Comment by sunnamed434** [comment]
It might be about that newer dotnet versions compiles into native code directly, what's the target framework of your file?

**Comment by Sicarmon** [comment]
The target framework of the file is .NET 4.0. It could be correctly decompiled before installing Visual Studio 2022, but the problem appeared after the installation. Additionally, I can decompile it using dotPeek, but the code from dotPeek is not as good as that from dnSpy. I believe the decompiled code from dnSpy is very close to the original source code. I highly suspect that the installation of VS2022 has somehow affected the environment for dnSpy. Tomorrow, I will try testing on a VPS to see if dnSpy can correctly decompile the target file without VS2022 installed. I will update you with the test results then. Thank you!  



---- Replied Message ----
| From | ***@***.***> |
| Date | 9/21/2024 03:54 |
| To | ***@***.***> |
| Cc | ***@***.***>,
***@***.***> |
| Subject | Re: [dnSpyEx/dnSpy] Only PE After System install Visual Studio 2022 (Discussion #362) |

It might be about that newer dotnet versions compiles into native code directly, what's the target framework of your file?

**Comment by ElektroKill** [comment]
Without access to the file I cannot assess the problem. Please provide it to me on github or via email elektrokilldev@protonmail.com
